react/features/base/i18n/languageDetector.native.js

```diff
@@ -17,7 +17,7 @@ export default {
     detect() {
         const { LocaleDetector } = NativeModules;
 
-        return LocaleDetector.locale.replace(/_/, '-');
+        return LocaleDetector.locale.replace(/[_-]/, '');
```

Comment:
Can you please explain why this is needed?

Reply:
we got lngWhitelist, whitelist:['en','zhCN','zhTW'...] from languages.json used by 'i18next'. on my android phone, the native module LocalDetector returned the language code 'zh_CN' which is changed to 'zh-CN' in js due to the upon code, and it is splitted into  ['zh', 'CN'] in i18next.  only the first part 'zh' is treated as the language code,and it is not in the whitelist, so...

i deleted '-'/'_' from language code in order to make it work. im not sure whether is  is a good practice for use of 'i18next'.

Reply:
i think it is a problem of resolving language resources.

while the language code is 'zhCN' and the language file name is 'main-zhCN.json',  the localeDetector module got 'zh_CN' which means language code is 'zh_CN' and will point to the file 'lang/main-zh_CN.json', and it dose not corrspond to the language options.

mybe there are different results in other phones like 'zh_CN','zh-CN' when the locale language is Chinese, and this replacement is just an alias. or it tries to resolve a sub option of Simplified Chinese or Traditional Chinese.

Reply:
I see. Here is how we define it: https://github.com/jitsi/jitsi-meet/blob/master/react/features/base/i18n/BuiltinLanguages.native.js#L137 as you can see it's "zhCN". So the problem seems to be indeed in the detector.

Your change would also affect iOS though, have you tested it there?

Reply:
code in LocaleDetector.java : context.getResources().getConfiguration().locale.toString() results in a '_' seperated locale string(languageCode_regionCode, i got 'zh_CN'), it is correct that we replace '_' with '-' in order to let 'i18next' distinguish the language code(iso-639-1) part from locale string in js.

'zh' is language code and 'CN' is region code.

when init 'i18next'  with option: load = 'unspecific',  the option will be overrided by default value: 'languageOnly'. and it will only use the language code part to set the language. e.g 'zh-CN': 'languageOnly' -> 'zh', 'currentOnly' -> 'zh-CN','all'->'zh','zh-CN'. and 'zh' is not in the language white list.

for most of the time, we can just use the language code splitted by '-' from the locale string.

we should configure 'language.json', replace 'zhCN' with 'zh', add 'zhTW' to make 'zh', 'zhTW' in the white list, do the same thing with BuiltinLanguages.native.js to add resource bundle to 'i18next' with name 'zh','zhTW', and then set the init option 'load' to 'all' .

so we can lookup Traditional Chinese translations by 'zh, zhTW' when the locale string is 'zh_TW', Simplified Chinese translations by 'zh' when the locale string is 'zh_CN'.

in ios, localDetector returns '-' seperated locale string, but it seems that we may get 'zh-Hans'/'zh-Hant' but not 'zh-CN' or 'zh-TW'.  which results only looking up translations of Simplified Chinese(zh). i think we should make a special treat for converting the local string to 'zh-CN'/'zh-TW' anywhere in localeDetector.m or languageDetector.native.js.

i will create a new PR later.